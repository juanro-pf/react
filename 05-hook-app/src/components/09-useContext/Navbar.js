import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <Link className="navbar-brand" to="/">useContext</Link>
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink activeClassName='active' exact to="/" className="nav-link" >Home</NavLink>
                    <NavLink activeClassName='active' exact to="/login" className="nav-link" >Login</NavLink>
                    <NavLink activeClassName='active' exact to="/about" className="nav-link" >About</NavLink>
                </div>
            </div>
        </nav>
    );
};
