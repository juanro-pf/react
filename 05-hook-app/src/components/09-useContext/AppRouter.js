import React from 'react';
import { Navbar } from './Navbar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";

export const AppRouter = () => {
    return (
        <div>
            <Router>
                <div>

                    <Navbar />

                    <div className='container'>
                        <Switch>
                            <Route exact path='/' component= { HomeScreen } />
                            <Route exact path='/login' component= { LoginScreen } />
                            <Route exact path='/about' component= { AboutScreen } />
                            {/* <Route component= { HomeScreen } /> */}
                            <Redirect to='./' />
                        </Switch>
                    </div>

                </div>
            </Router>
        </div>
    );
};
