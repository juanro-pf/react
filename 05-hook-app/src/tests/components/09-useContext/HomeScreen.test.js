import React from 'react';
import { shallow, mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

import '@testing-library/jest-dom';

describe('Testing HomeScreen', () => {

    const user= {
        name: 'Juan Ro',
        email: 'email'
    }
    
    //mount se usa para ver tambien los componentes dentro de componentes (HomeScreen dentro de Usecontext.Provider)
    const wrapper= mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('Snapshot', () => {

        expect(wrapper).toMatchSnapshot();

    });
    

});
