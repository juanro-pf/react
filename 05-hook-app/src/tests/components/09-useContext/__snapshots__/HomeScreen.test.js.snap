// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeScreen Snapshot 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
  "name": "Juan Ro",
  "email": "email"
}
  </pre>
</div>
`;
